### PR TITLE
Switch `dev-ui` command to use `npm run dev`

### PR DESCRIPTION
**Pull request by dr-frmr**
Frontend build command should use a fast debug build unless developer specifically wants to build for release. To this end, we switch `dev-ui` to use `npm run dev` unless the `--release` flag is passed, in which case it will use `npm start`.

`kit b` will remain unchanged.

### CODE REVIEW

**Comment by nick1udwig** [comment]
Comment in general before diving in to the review: we don't currently enforce rustfmt on this repo and I don't want to until I can look into how to make it not make certain cases less readable IMO, see #74 

However, I know you and some others have editors configured to auto-rustfmt, and that makes it harder to review PRs because there are lots of lines changed that are just formatting. Is there a way you can toggle this behavior so you are not making lots of formatting changes here/in future?

File: src/main.rs [hunk 1]
```diff
@@ -504,8 +514,13 @@ async fn make_app(current_dir: &std::ffi::OsString) -> anyhow::Result<Command> {
             )
         )
         .subcommand(Command::new("dev-ui")
-            .about("Start the web UI development server with hot reloading (same as `cd ui && npm i && npm start`)")
+            .about("Start the web UI development server with hot reloading (same as `cd ui && npm i && npm run dev`). To instead create a production build, use the --release flag.")
```
Comment (nick1udwig) [inline src/main.rs:517]: This helptext is getting pretty long. Can we keep it as it was and just change the `npm start` to `npm run dev`?

File: src/main.rs [hunk 2]
```diff
@@ -504,8 +514,13 @@ async fn make_app(current_dir: &std::ffi::OsString) -> anyhow::Result<Command> {
...
             .visible_alias("d")
+            .arg(Arg::new("RELEASE")
+                .action(ArgAction::SetTrue)
+                .long("release")
+                .help("If set, create a production build")
+            )
```
Comment (nick1udwig) [inline src/main.rs:523]: This should probably come after at least some of the other args since this affects the print order of the helptext.

**Review by nick1udwig — APPROVED** [review]
Added some comments. Please address, then good to merge. Thanks for this work!

**Review by gdbroman — APPROVED** [review]
Tested this with fake-node and it does what was intended 👍

File: src/main.rs (hunk 2 above)
Comment (nick1udwig) [reply src/main.rs]: [f3d1590](https://github.com/kinode-dao/kit/pull/112/commits/f3d15901d642319c59e048b2d0671f3170653801)

File: src/main.rs (hunk 1 above)
Comment (nick1udwig) [reply src/main.rs]: [f3d1590](https://github.com/kinode-dao/kit/pull/112/commits/f3d15901d642319c59e048b2d0671f3170653801)

File: src/dev_ui/mod.rs [hunk 3]
```diff
@@ -21,28 +26,45 @@ pub fn execute(package_dir: &Path, url: &str, skip_deps_check: bool) -> anyhow::
...
+                .args(&["-c", &install])
+                .current_dir(&ui_path),
         )?;
 
-        info!("Running npm start...");
+        if release {
+            info!("Running npm start");
+        } else {
+            info!("Running npm run dev...");
+        }
```
[the remark this answers is not in the document]
Comment (nick1udwig) [reply src/dev_ui/mod.rs]: [f3d1590](https://github.com/kinode-dao/kit/pull/112/commits/f3d15901d642319c59e048b2d0671f3170653801)

**Pull request merged**